pandas/core/indexes/period.py

```diff
@@ -503,7 +503,7 @@ def __array_wrap__(self, result, context=None):
 
     @property
     def _formatter_func(self):
-        return lambda x: "'%s'" % x
+        return self.array._formatter(boxed=False)
```

Comment:
why is this the only index sublcass that you need to do this for?

Reply:
It's the only extension-array backed index that's using the formatting right now.

- CategoricalIndex always uses the default formatter for the underlying categories (since it can be a container for any type, it dispatches the formatting).
- IntervalIndex /IntervalArray don't use this formatting

Datetime / TImedelta will use this too, so this will be pushed up into `DatetimeIndexOpsMixin` later.